spec/ruby/optional/capi/string_spec.rb

```diff
@@ -1227,4 +1227,36 @@ def inspect
       -> { @s.rb_str_unlocktmp("test") }.should raise_error(RuntimeError, 'temporal unlocking already unlocked string')
     end
   end
+
+  describe "rb_enc_interned_str_cstr" do
+    it "returns a frozen string" do
+      str = "hello\0not used"
+      val = @s.rb_enc_interned_str_cstr(str, Encoding::US_ASCII)
+
+      val.should.is_a?(String)
+      val.encoding.should == Encoding::US_ASCII
+      val.should.frozen?
+    end
+
+    it "returns the same frozen string" do
+      str = "hello\0not used"
+      val1 = @s.rb_enc_interned_str_cstr(str, Encoding::US_ASCII)
+      val2 = @s.rb_enc_interned_str_cstr(str, Encoding::US_ASCII)
+
+      val1.object_id.should == val2.object_id
+
+      val1 = @s.rb_enc_interned_str_cstr(str, Encoding::US_ASCII)
+      val2 = @s.rb_enc_interned_str_cstr(str, Encoding::UTF_8)
+
+      val1.object_id.should != val2.object_id
+    end
+
+    # TODO: This doesn't work in MRI because https://github.com/ruby/ruby/blob/8ba8e979c8b8a71dcc308c392bcead38af5306c7/string.c#L12067-L12069
+    it "uses the default encoding when enc is a null pointer" do
```

Comment:
What happens when you run on MRI? For what it's worth, I think it may just be a documentation issue in CRuby that we can fix.

Reply:
It's a segmentation fault. We'll remove the null handling for now and investigate fixing or updating the documentation in CRuby.